pyo3-build-config/src/impl_.rs

```diff
@@ -733,39 +828,76 @@ pub(crate) fn cross_compile_env_vars() -> CrossCompileEnvVars {
 ///   `PYO3_CROSS_LIB_DIR`.
 ///
 /// See the [PyO3 User Guide](https://pyo3.rs/) for more info on cross-compiling.
+#[deprecated(
+    since = "0.17.0",
+    note = "please use cross_compiling_from_to() instead"
+)]
 pub fn cross_compiling(
     host: &str,
     target_arch: &str,
     target_vendor: &str,
     target_os: &str,
 ) -> Result<Option<CrossCompileConfig>> {
-    let env_vars = cross_compile_env_vars();
+    let host: Triple = host.parse().map_err(|_| "bad host triple")?;
+
+    let architecture: Architecture = target_arch.parse().map_err(|_| "bad target arch")?;
+    let vendor: Vendor = target_vendor.parse().map_err(|_| "bad target vendor")?;
+    let operating_system: OperatingSystem = target_os.parse().map_err(|_| "bad target os")?;
 
-    let target_info = TargetInfo {
-        arch: target_arch.to_owned(),
-        vendor: target_vendor.to_owned(),
-        os: target_os.to_owned(),
+    // FIXME: This is a very bad approximation that only works
+    // for the current `CrossCompileConfig` implementation.
+    let environment = match operating_system {
+        OperatingSystem::Windows => Environment::Msvc,
+        _ => Environment::Gnu,
     };
 
-    if !env_vars.any() && is_not_cross_compiling(host, &target_info) {
-        return Ok(None);
-    }
+    // FIXME: This field is currently unused.
```

Comment:
I guess we don't need to address these `FIXME`s right now since we're going to remove `cross_compiling` in a new minor version.

Reply:
Oh, certainly. These FIXMEs will be removed with `cross_compiling()` in v0.17.
Perhaps I should've used `WONTFIX:` instead of `FIXME:` :grin:.